src/wire/tcp.rs

```diff
@@ -553,6 +595,47 @@ impl<'a> TcpOption<'a> {
                         option = TcpOption::WindowScale(data[0]),
                     (field::OPT_WS, _) =>
                         return Err(Error::Malformed),
+                    (field::OPT_SACKPERM, 2) =>
+                        option = TcpOption::SackPermitted,
+                    (field::OPT_SACKPERM, _) =>
+                        return Err(Error::Malformed),
+                    (field::OPT_SACKRNG, n) => {
+                        if n < 10 || (n-2) % 8 != 0 {
+                            return Err(Error::Malformed)
+                        }
+                        if n > 26 {
+                            // It's possible for a remote to send 4 SACK blocks, but extremely rare.
+                            // Better to "lose" that 4th block and save the extra RAM and CPU
+                            // cycles in the vastly more common case.
```

Comment:
This seems like a weird optimization, but since the RFC spells it out explicitly, I guess it's fine.